utils/python/CIME/XML/Machines.py

```diff
@@ -0,0 +1,118 @@
+        in the NODENAME_REGEX field in the file.   First match wins.
+        """
+        nodes = self.get_node('machine')
+        for node in nodes:
+            machine = node.get('MACH')
+            logging.debug("machine is "+machine)
+            self.set_machine(machine)
+            regex_str_nodes =  self.get_node('NODENAME_REGEX',root=self.machine)
+            if(len(regex_str_nodes)>0):
+                regex_str = regex_str_nodes[0].text
+                logging.debug("machine regex string is "+ regex_str)
+                if (regex_str is not None):
+                    regex = re.compile(regex_str)
+                    if (regex.match(nametomatch)):
+                        return machine
+
+        return None
+
+
+    def set_machine(self,machine):
+        """
+        Sets the machine block in the Machines object
+        """
+        self.machine = self.get_node('machine',{'MACH':machine})[0]
+        expect(self.machine is not None,"No machine %s found" % machine)
+                
+        self.name = machine
+
+    def get_value(self,name):
+        """
+        Get Value of fields in the config_machines.xml file
+        """
+        expect(self.machine is not None, "Machine object has no machine defined")
+        value = None
+        """
+        COMPILER and MPILIB are special, if called without arguments they get the default value from the
+        COMPILERS and MPILIBS lists in the file. 
+        """
+        if(name == "COMPILER"):
+            value = self.get_compiler()
+        elif(name == "MPILIB"):
+            value = self.get_MPIlib()
+        else:
+            nodes = self.get_node(name,root=self.machine)
+            if(len(nodes)>0):
+                node = nodes[0]
+                expect(node is not None,"No match found for %s in machine %s" % (name,self.name))
+                value = node.text
+        if(value is None):
+            """ if all else fails """
+            value = GenericXML.get_value(self,name)
+        return value
+
+    def get_field_from_list(self, listname, reqval=None):
+        """ 
+        Some of the fields have lists of valid values in the xml, parse these
+        lists and return the first value if reqval is not provided and reqval 
+        if it is a valid setting for the machine
+        """
+
+        expect(self.machine is not None, "Machine object has no machine defined")
+        supported_values = self.get_value(listname)
+        if(supported_values is None):
+            logging.critical("No list found for "+listname+" on machine "+self.name)
+        supported_values = supported_values.split(',')
+        if(reqval is None or reqval == "UNSET"):
+            return supported_values[0]
+        for val in supported_values:
+            if(val == reqval):
+                return reqval
+        logging.critical("%s value %s not supported for machine %s" %
+                         (listname, reqval, self.name))
+
+    def get_compiler(self, compiler=None):
```

Comment:
I'm a little confused here, why would a user do get_compiler instead of get_value("compiler")? Is it just a convenience thing because we expect them to be asking for this a lot?